src/CreateMiddleware/CreateMiddleware.php

```diff
@@ -41,10 +41,11 @@ public function process(ServerRequestInterface $request, DelegateInterface $dele
     /**
      * @param string $class
      * @param string|null $projectRoot
+     * @param string|null $classSkeleton
      * @return string
      * @throws CreateMiddlewareException
      */
-    public function process($class, $projectRoot = null)
+    public function process($class, $projectRoot = null, $classSkeleton = null)
```

Comment:
This package is still not released as stable, so we'll create version `0.5.0`. Problem solved :D

On a serious note: What I'm trying to say is if you change self:: to static:: you need to extend the CreateMiddleware class **and** you need to write a custom command. If you inject the class skeleton you can reuse the original CreateMiddleware class for multiple purposes and only need to create a custom commands.

In my case I have 3 templates. api, html (HtmlResponse) and normal Middleware. So if I can inject the class skeleton I have to write _only_ 3 command classes. Otherwise I need to extend the CreateMiddleware class 3 times _and_ write 3 command classes.